app/script/event/EventService.js

```diff
@@ -0,0 +1,160 @@
+z.event.EventService = class EventService {
+  /**
+   * Construct a new Event Service.
+   * @param {StorageService} storageService - Service for all storage interactions
+   */
+  constructor(storageService) {
+    this.storageService = storageService;
+    this.logger = new z.util.Logger('z.conversation.EventService', z.config.LOGGER.OPTIONS);
+    this.EVENT_STORE_NAME = z.storage.StorageSchemata.OBJECT_STORE.EVENTS;
+  }
+
+  /**
+   * Load event from DB
+   *
+   * @param {string} conversationId - ID of conversation
+   * @param {string} eventId - ID of event to retrieve
+   * @returns {Promise} Resolves with the stored record
+   */
+  loadEvent(conversationId, eventId) {
+    if (!conversationId || !eventId) {
+      this.logger.error(`Cannot get event '${eventId}' in conversation '${conversationId}' without IDs`);
+      const error = new z.conversation.ConversationError(z.conversation.ConversationError.TYPE.MISSING_PARAMETER);
+      return Promise.reject(error);
+    }
+
+    return this.storageService.db[this.EVENT_STORE_NAME]
+      .where('id')
+      .equals(eventId)
+      .filter(record => record.conversation === conversationId)
+      .first()
+      .catch(error => {
+        const logMessage = `Failed to get event '${eventId}' for conversation '${conversationId}': ${error.message}`;
+        this.logger.error(logMessage, error);
+        throw error;
+      });
+  }
+
+  /**
+   * Get events with given category.
+   *
+   * @param {string} conversationId - ID of conversation to add users to
+   * @param {MessageCategory} categoryMin - Minimum message category
+   * @param {MessageCategory} [categoryMax=z.message.MessageCategory.LIKED] - Maximum message category
+   * @returns {Promise} Resolves with matching events
+   */
+  loadEventsWithCategory(conversationId, categoryMin, categoryMax = z.message.MessageCategory.LIKED) {
+    return this.storageService.db[this.EVENT_STORE_NAME]
+      .where('[conversation+category]')
+      .between([conversationId, categoryMin], [conversationId, categoryMax], true, true)
+      .sortBy('time');
+  }
+
+  /**
+   * Load conversation events starting from the upper bound going back in history
+   *  until either limit or lower bound is reached.
+   *
+   * @param {string} conversationId - ID of conversation
+   * @param {Date} [lowerBound=new Date(0)] - Load from this date (included)
+   * @param {Date} [upperBound=new Date()] - Load until this date (excluded)
+   * @param {number} [limit=Number.MAX_SAFE_INTEGER] - Amount of events to load
+   * @returns {Promise} Resolves with the retrieved records
+   */
+  loadPrecedingEvents(
+    conversationId,
+    lowerBound = new Date(0),
+    upperBound = new Date(),
+    limit = Number.MAX_SAFE_INTEGER
+  ) {
+    if (!_.isDate(lowerBound) || !_.isDate(upperBound)) {
+      throw new Error(
+        `Lower bound (${typeof lowerBound}) and upper bound (${typeof upperBound}) must be of type 'Date'.`
+      );
```

Comment:
Can we clean up the code a bit while we are at it? Let's move the error message into a const and turn three lines into two here and the few other instances in this and the compount service file.